engine/src/witness/sol.rs

```diff
@@ -114,16 +110,12 @@ impl VoterApi<SolanaNonceTracking> for SolanaNonceTrackingVoter {
 		properties: <SolanaNonceTracking as ElectoralSystem>::ElectionProperties,
 	) -> Result<<<SolanaNonceTracking as ElectoralSystem>::Vote as VoteStorage>::Vote, anyhow::Error>
 	{
-		let (nonce_account, _previous_nonce) = properties;
-		let (response_account, response_nonce) =
-			nonce_witnessing::get_durable_nonces(&self.client, vec![nonce_account])
-				.await?
-				.pop()
-				.expect("If the query succeeds, we expect a nonce for the account we queried for");
-
-		assert_eq!(response_account, nonce_account);
-
-		Ok(response_nonce)
+		let (nonce_account, previous_nonce) = properties;
+		Ok(nonce_witnessing::get_durable_nonce(&self.client, nonce_account)
+			.await?
+			// If the nonce is not found, we default to the previous nonce.
+			// The `Change` electoral system ensure this vote is filtered.
+			.unwrap_or(previous_nonce))
```

Comment:
Wouldn't be rather not vote? We'll get like 150*10 extrinics every 6 seconds for no reason.

Reply:
As per the comment - this is dealt with by the implementation of the 'change' electoral system. We vote for the 'latest' value and if it is unchanged, the vote is filtered out and effectively ignored. This is not super obvious, I agree. I would prefer if it was more explicit and we returned, say, `None`. But I think returning the current nonce is the correct thing to do here, given the expected behaviour of the electoral system. And changing it to Option<Vote> would be a much deeper change, requiring migrations. I would rather avoid this until we have a better understanding of the elections pallet.